context/browsercontroller_client.go

```diff
@@ -480,3 +484,45 @@ func (rc *RecordContext) notifyDataReceived(activity browsercontroller.NotifyAct
 	}
 	return err
 }
+
+func RegisterConnectRequest(ctx filters.Context, conn *serviceconnections.Connections, proxyId int32, uri *url.URL) (jid, eid string, cid *config.ConfigRef) {
+	l := LogWithContext(ctx, "PROXY:BCC")
+
+	bccRequest := &browsercontroller.DoRequest{
+		Action: &browsercontroller.DoRequest_New{
+			New: &browsercontroller.RegisterNew{
+				ProxyId: proxyId,
+				Method:  "CONNECT",
+				Uri:     uri.String(),
+			},
+		},
+	}
+
+	bccCtx, cancel := context.WithCancel(context.Background())
```

Comment:
IMHO calling `defer cancel()` right after creating the context is more idiomatic than calling `cancel()` before every return.